ui/src/flux/components/FieldList.tsx

```diff
@@ -0,0 +1,150 @@
+
+// Utils
+import parseValuesColumn from 'src/shared/parsing/flux/values'
+import {
+  notifyCopyToClipboardSuccess,
+  notifyCopyToClipboardFailed,
+} from 'src/shared/copy/notifications'
+
+// types
+import {Source, NotificationAction} from 'src/types'
+
+interface Props {
+  db: string
+  source: Source
+  tag?: {key: string; value: string}
+  notify: NotificationAction
+}
+
+interface State {
+  fields: string[]
+  searchTerm: string
+}
+
+class FieldList extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      fields: [],
+      searchTerm: '',
+    }
+  }
+
+  public async componentDidMount() {
+    const fields = await this.fetchFields()
+    this.setState({fields})
+  }
+
+  public render() {
+    const {tag} = this.props
+    const {searchTerm} = this.state
+
+    return (
+      <>
+        <div className="flux-schema--filter">
+          <input
+            className="form-control input-xs"
+            placeholder={`Filter within ${(tag && tag.value) || 'Fields'}`}
+            type="text"
+            spellCheck={false}
+            autoComplete="off"
+            value={searchTerm}
+            onClick={this.handleInputClick}
+            onChange={this.onSearch}
+          />
+        </div>
+        {this.fields}
+      </>
+    )
+  }
+
+  private get fields(): JSX.Element | JSX.Element[] {
+    const {searchTerm} = this.state
+
+    const term = searchTerm.toLocaleLowerCase()
+    const fields = this.state.fields.filter(f =>
+      f.toLocaleLowerCase().includes(term)
+    )
+
+    if (fields.length) {
+      return fields.map(field => (
+        <div
+          className="flux-schema-tree flux-schema--child"
+          key={field}
+          onClick={this.handleClick}
+        >
+          <div className="flux-schema--item">
+            <div className="flex-schema-item-group">
+              {field}
+              <span className="flux-schema--type">Field</span>
+            </div>
+            <CopyToClipboard text={field} onCopy={this.handleCopyAttempt}>
+              <div className="flux-schema-copy" onClick={this.handleClick}>
+                <span className="icon duplicate" title="copy to clipboard" />
+                Copy
+              </div>
+            </CopyToClipboard>
+          </div>
+        </div>
+      ))
+    }
+
+    return (
+      <div className="flux-schema-tree flux-schema--child">
+        <div className="flux-schema--item no-hover" onClick={this.handleClick}>
+          <div className="no-results">No more fields.</div>
+        </div>
+      </div>
+    )
+  }
+
+  private async fetchFields(): Promise<string[]> {
+    const {source, db, tag} = this.props
+
+    const filter = !!tag ? [tag] : []
+    const limit = 50
+
+    try {
+      const response = await fetchFields(source, db, filter, limit)
+      const fields = parseValuesColumn(response)
+      return fields
+    } catch (error) {
+      console.error(error)
```

Comment:
If the fields fail to fetch, then this will silently fail and the user will just see an empty list. How about instead we either display a notification, or wrap the component in `ErrorHandling` and let the exception throw?